src/promptflow/promptflow/_core/operation_context.py

```diff
@@ -111,14 +111,38 @@ def get_user_agent(self):
 
         def parts():
             # Be careful not to use "self.user_agent" as it will recursively call get_user_agent in __getattr__
-            agent = self.get(self._USER_AGENT, '')
+            agent = self.get(self._USER_AGENT, "")
             yield agent
             promptflow_agent = f"promptflow/{VERSION}"
-            yield promptflow_agent if promptflow_agent not in agent else ''
-            user_agent = os.environ.get(USER_AGENT, '').strip()
-            yield user_agent if user_agent not in agent else ''
-            pf_user_agent = os.environ.get(PF_USER_AGENT, '').strip()
-            yield pf_user_agent if pf_user_agent not in agent else ''
+            yield promptflow_agent if promptflow_agent not in agent else ""
+            user_agent = os.environ.get(USER_AGENT, "").strip()
```

Comment:
I'm a little confused why the environment variable loading logic is here, and what is the meaning and the differences among "agent", "USER_AGENT" and "PF_USER_AGENT"?

Reply:
This part is rather confusing. I don't understand why we add logic like this here from glimpse of code. Any context?

Reply:
Moved out to separate file